backend/src/schema/resolvers/notifications.js

```diff
@@ -48,16 +48,20 @@ export default {
       const readTxResultPromise = session.readTransaction(async transaction => {
         const notificationsTransactionResponse = await transaction.run(
           ` 
-          MATCH (resource {deleted: false, disabled: false})-[notification:NOTIFIED]->(user:User {id:$id})
-          ${whereClause}
-          WITH user, notification, resource,
-          [(resource)<-[:WROTE]-(author:User) | author {.*}] as authors,
-          [(resource)-[:COMMENTS]->(post:Post)<-[:WROTE]-(author:User) | post{.*, author: properties(author)} ] as posts
-          WITH resource, user, notification, authors, posts,
-          resource {.*, __typename: labels(resource)[0], author: authors[0], post: posts[0]} as finalResource
-          RETURN notification {.*, from: finalResource, to: properties(user)}
-          ${orderByClause}
-          ${offset} ${limit}
+            MATCH (resource)-[notification:NOTIFIED]->(user:User {id:$id})
+            WHERE
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-01-23T18:47:26Z" title="Thursday, January 23rd 2020, 7:47:26 pm +01:00">Jan 23, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -48,16 +48,18 @@ export default {
       const readTxResultPromise = session.readTransaction(async transaction => {
         const notificationsTransactionResponse = await transaction.run(
           ` 
-          MATCH (resource {deleted: false, disabled: false})-[notification:NOTIFIED]->(user:User {id:$id})
-          ${whereClause}
-          WITH user, notification, resource,
-          [(resource)<-[:WROTE]-(author:User) | author {.*}] as authors,
-          [(resource)-[:COMMENTS]->(post:Post)<-[:WROTE]-(author:User) | post{.*, author: properties(author)} ] as posts
-          WITH resource, user, notification, authors, posts,
-          resource {.*, __typename: labels(resource)[0], author: authors[0], post: posts[0]} as finalResource
-          RETURN notification {.*, from: finalResource, to: properties(user)}
-          ${orderByClause}
-          ${offset} ${limit}
+            MATCH (resource)-[notification:NOTIFIED]->(user:User {id:$id})
+            WHERE ((labels(resource)[0] in ["Post", "Comment"] AND NOT resource.deleted AND NOT resource.disabled) OR labels(resource)[0] in ["Report"])
```


cool stuff! maybe the first time we've added something like this to the code base?

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-01-24T09:33:55Z" title="Friday, January 24th 2020, 10:33:55 am +01:00">Jan 24, 2020</time>_

---


Love that you like it!
Yes, I think so …